src/internal/connector/onedrive/collections.go

```diff
@@ -434,24 +436,53 @@ func (c *Collections) UpdateCollections(
 		var (
 			prevPath           path.Path
 			prevCollectionPath path.Path
+			ok                 bool
 		)
 
 		if item.GetRoot() != nil {
-			// Skip the root item
+			rootPath, err := GetCanonicalPath(
+				fmt.Sprintf(rootDrivePattern, driveID),
+				c.tenant,
+				c.resourceOwner,
+				c.source,
+			)
+			if err != nil {
+				return err
+			}
+
+			updatePath(newPaths, *item.GetId(), rootPath.String())
```

Comment:
Just checking, this is called to add the folder to the paths map? If so, could you add a comment to this line saying that. It's a little confusing off the cuff because I wouldn't expect the path of the root folder to change

Reply:
Yup, that is it. I'll add a comment.